src/Microsoft.OData.Edm/Csdl/CsdlLocation.cs

```diff
@@ -26,6 +26,11 @@ internal CsdlLocation(string source, int number, int position)
             this.LinePosition = position;
         }
 
+        internal CsdlLocation(string jsonPath)
+        {
+            JsonPath = jsonPath;
+        }

```

Comment:
Might be better to define a JsonCsdlLocation : CsdlLocation.

Reply:
I did the same thing at the beginning, but i revert it. because if we have the JsonCsdlLocation, we should have a CsdlXmlLocation?

Besides, CsdlLocation should be "internal" class (I don't know why it's public).

It's open for discuss. 



---
In reply to: [429387626](https://github.com/OData/odata.net/pull/1775#discussion_r429387626) [](ancestors = 429387626)